### PR TITLE
HL-820 | P2P / Ahjo inspection inputs are shown/hidden based on radio button's value

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

Only Ahjo **OR** P2P fields are needed, depending on case. Only display form fields based on which one is better for koonti at hand.

* Renewed colors and design
* Only either / or is required from formik and yup
* Change backend validation rules

screencapture-localhost-3100-batches-2023-08-07-15_41_07

### CODE REVIEW

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe employer result is __failed__ for https://kesaseteli-pr2165.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by sirtawast** [comment]
> Changed model but no migration file? And some linters are complaining. Otherwise, looks good!

`$ python manage.py makemigrations` results to *"No changes detected"*.

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2165.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2165.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe employer result is __failed__ for https://kesaseteli-pr2165.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2165.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/handler/src/components/batchProcessing/BatchApplicationList.tsx [hunk 1]
```diff
@@ -217,7 +227,16 @@ const BatchApplicationList: React.FC<BatchProps> = ({ batch }: BatchProps) => {
               initialSortingOrder="asc"
               cols={cols}
             />
-            <$TableFooter>
+            <$TableFooter
+              backgroundColor={
+                [
```
Comment (mjturt) [inline frontend/benefit/handler/src/components/batchProcessing/BatchApplicationList.tsx:232]: Same array literal used 2 times. Not a big deal.

File: frontend/benefit/handler/src/components/batchProcessing/BatchActionsInspectionForm.tsx [hunk 2]
```diff
@@ -33,7 +42,8 @@ const BatchActionsInspectionForm: React.FC<BatchProps> = ({
   isInspectionFormSent,
   setInspectionFormSent,
   setBatchCloseAnimation,
-}: BatchProps) => {
+}: // eslint-disable-next-line sonarjs/cognitive-complexity
```
[the remark this answers is not in the document]
Comment (sirtawast) [reply frontend/benefit/handler/src/components/batchProcessing/BatchActionsInspectionForm.tsx:45]: This process will be automated in the future so I thought "keep it simple, stupid".

[the remark this answers is not in the document]
Comment (sirtawast) [reply frontend/benefit/handler/src/components/batchProcessing/BatchActionsInspectionForm.tsx:45]: Good catch though! I was thinking Sonar would whine about the props itself but maybe it's the whole component 🤔

**Pull request merged**